plan_delegator/src/plan_delegator.cpp

```diff
@@ -730,21 +732,41 @@ namespace plan_delegator
             trajectory_plan.header.stamp = get_clock()->now();
             last_successful_traj_ = trajectory_plan;
             traj_pub_->publish(trajectory_plan);
+            consecutive_traj_gen_failure_num_ = 0;
         }
         else
         {
+            consecutive_traj_gen_failure_num_ ++;
             RCLCPP_WARN_STREAM(rclcpp::get_logger("plan_delegator"),
                 "Guidance is engaged, but new planned trajectory has less than 2 points. " <<
-                "It will not be published!");
-            if (last_successful_traj_.has_value())
+                "It will not be published! Consecutive failure count: "
+                << consecutive_traj_gen_failure_num_);
+
+            // case where traj generation fails after a successful one
+            if (last_successful_traj_.has_value()
+                && consecutive_traj_gen_failure_num_
+                    <= config_.max_traj_generation_reattempt)
             {
                 RCLCPP_WARN_STREAM(rclcpp::get_logger("plan_delegator"),
-                    "Instead, last available trajectory is published with outdated timestamp of:" <<
-                    std::to_string(
+                    "Instead, last available trajectory is published with outdated timestamp of:"
+                    << std::to_string(
                         rclcpp::Time(last_successful_traj_.value().header.stamp).seconds()));
                 traj_pub_->publish(last_successful_traj_.value());
             }
-
+            // case where traj generation fails from the beginning
+            else if (!last_successful_traj_.has_value() &&
+                consecutive_traj_gen_failure_num_ <= config_.max_traj_generation_reattempt)
+            {
+                RCLCPP_WARN_STREAM(rclcpp::get_logger("plan_delegator"),
+                    "Instead, tried publishing last available trajectory, but it is not available!");
+            }
```

Comment:
This should be throwing an error as well? Do we need this to be different from the else condition?

Reply:
Discussed offline, not throwing an error here because we want a minimum number of replans before throwing an error.